### PR TITLE
Proc_macro client impls and generate functionality

### PR DESCRIPTION
**Pull request by AloeareV**

### CODE REVIEW

**Comment by zancas** [comment]
Is this still the thing I should review first, or should I look at the other PR?

File: zcashrpc-macros/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,68 @@
...
+        call_ident_string.insert(1, '_');
+    }
+    let call_ident = proc_macro2::Ident::new(&call_ident_string, ident.span());
+    let response_ident = proc_macro2::Ident::new(
+        &format!("{}Response", ident.to_string()),
+        ident.span(),
+    );
+    let typed_args_as_stream = typed_args.stream();
+    let untyped_args_as_stream: proc_macro2::TokenStream =
+        typed_args.stream().into_iter().step_by(4).collect();
```
[the remark this answers is not in the document]
Comment (zancas) [reply zcashrpc-macros/src/lib.rs:62]: I am in agreement about replacing the magical `step_by`, with something more explicit, and renaming `untyped_args_as_stream`.

**Review by zancas — CHANGES_REQUESTED** [review]

File: zcash-rcli/src/commands.rs [hunk 2]
```diff
@@ -51,20 +57,26 @@ fn make_client(regtest: bool) -> zcashrpc::Client {
...
+
             #[options(help = "command-specific help")]
             help: bool,
         }
         impl Runnable for $command {
             fn run(&self) {
                 abscissa_tokio::run(&$crate::application::APPLICATION, async {
                     let response =
-                        $crate::commands::make_client(true).$rpc_call();
+                        $crate::commands::make_client(true).$rpc_call($($(self.$arg),+)?);
```
[the remark this answers is not in the document]
Comment (AloeareV) [reply zcash-rcli/src/commands.rs:79]: Correct

**Comment by AloeareV** [comment]
This is made obsolete by pr #13. Closing.

**Pull request closed without merging**